core/src/main/scala/pureconfig/ConfigCursor.scala

```diff
@@ -82,6 +82,21 @@ sealed trait ConfigCursor {
       case v => Left(CannotConvert(v.toString, "Long", "Unable to convert Number to Long"))
     })
 
+  /**
+   * Casts this cursor to a long representing a memory size according to the way com.typesafe.config.Config#getBytes(java.lang.String) does.
+   *
+   * @return a `Right` with the memory size value pointed to by this cursor if the cast can be done, `Left` with a list of
+   *         failures otherwise.
+   */
+  def asBytes: ConfigReader.Result[Long] = scopeFailure {
```

Comment:
I don't think this is suited for `ConfigCursor`. It's important to distinguish here between two different levels of config reading:
1. Primitive values, like integers or strings, that we can read out of `ConfigValues` because they are internally represented that way in HOCON;
1. Non-primitive values, represented in HOCON by one or more primitives in a known format.

`ConfigCursor` deals with the lower level, while the higher level is left to `ConfigReader`s. In this particular case we seem to be adding here code that is most suited to a `ConfigReader` (this is a very opinionated and high level way of reading bytes, in fact - `Config` tries to read this as a long, possibly falls back to reading a string and returns a `Long` instead of an `Array[Byte]`).

Note that the fact that it's supported by `Config` does _not_ make it a primitive - there are several other instances there unsuited to the cursor class, like `getMemorySize`, `getMilliseconds`, `getPeriod` and much more. Their API is bloated mainly because they don't have anything like configurable `ConfigReader`s.

Reply:
👍  since this pr also introduces a `ConfigReader` instance for `ConfigMemorySize` I'll just move this implementation there.